### PR TITLE
impl PartialOrd and some methods for Property

### PR DESCRIPTION
**Pull request by waruto210**
## What do these changes do?

The derived `PartialOrd` of enum is not suitable, impl PartialOrd and some methods for Property, then format it.

## Related issue number

None

### CODE REVIEW

File: interactive_engine/executor/store/src/api/property.rs [hunk 1]
```diff
@@ -44,6 +44,51 @@ pub enum Property {
...
+            // if both are integers, cast to long
+            // else cast to double
+            (Property::Short(_), _)
+            | (Property::Int(_), _)
+            | (Property::Long(_), _)
+            | (_, Property::Short(_))
+            | (_, Property::Int(_))
+            | (_, Property::Long(_)) => {
+                if self.is_float_type() || other.is_float_type() {
+                    let left = self.get_double().expect("Convert error");
```
Comment (BingqingLyu) [inline interactive_engine/executor/store/src/api/property.rs:62]: It's better not to use `expect()` here, which will panic once Property is neither integers nor floats. Same as the follows.

File: interactive_engine/executor/store/src/api/property.rs [hunk 2]
```diff
@@ -44,6 +44,51 @@ pub enum Property {
...
+            (Property::Float(left), Property::Float(right)) => left.partial_cmp(right),
+            (Property::Double(_), _) | (_, Property::Double(_)) => {
+                let left = self.get_double().expect("Convert error");
+                let right = other.get_double().expect("Convert error");
+                left.partial_cmp(&right)
+            }
+            (Property::Bytes(left), Property::Bytes(right)) => left.partial_cmp(right),
+            (Property::String(left), Property::String(right)) => left.partial_cmp(right),
+            (Property::Date(left), Property::Date(right)) => left.partial_cmp(right),
+            (Property::ListInt(left), Property::ListInt(right)) => left.partial_cmp(right),
```
Comment (BingqingLyu) [inline interactive_engine/executor/store/src/api/property.rs:80]: As above, we compare integers/floats by converting into `long`/`double`. How about compare between  e.g., `Property::ListInt` and `Property::ListLong`?

**Review by BingqingLyu — COMMENTED** [review]
As above.

File: interactive_engine/executor/store/src/api/property.rs (hunk 1 above)
Comment (waruto210) [reply interactive_engine/executor/store/src/api/property.rs]: introduce a macro to handle it

File: interactive_engine/executor/store/src/api/property.rs (hunk 2 above)
Comment (waruto210) [reply interactive_engine/executor/store/src/api/property.rs:84]: now cast number lists to cmp

**Comment by waruto210** [comment]
> As follows.

please review again

**Review by BingqingLyu — APPROVED** [review]
LGTM.

**Pull request merged**